packages/devtools_app/lib/src/debugger/codeview.dart

```diff
@@ -538,31 +549,96 @@ class Lines extends StatelessWidget {
         final lineNum = index + 1;
         return LineItem(
           lineContents: lines[index],
+          debuggerController: debuggerController,
           pausedFrame: pausedLine == lineNum ? pausedFrame : null,
         );
       },
     );
   }
 }
 
-class LineItem extends StatelessWidget {
+class LineItem extends StatefulWidget {
   const LineItem({
     Key key,
     @required this.lineContents,
+    @required this.debuggerController,
```

Comment:
access debugger controller from didChangeDependencies instead of requiring it to be a parameter for this stateful widget (see above comment for example)

Reply:
why didUpdateDependencies instead of the build method?